pkg/ingester/stream_rate_calculator.go

```diff
@@ -66,31 +71,79 @@ func (c *StreamRateCalculator) updateLoop() {
 }
 
 func (c *StreamRateCalculator) updateRates() {
-	rates := make([]logproto.StreamRate, 0, c.size)
+	rates := c.rates()
+	samples := c.currentSamplesPerTenant()
+	samples = updateSamples(rates, samples)
+
+	updatedRates := make([]logproto.StreamRate, 0, c.size)
```

Comment:
not sure if I'm missing something, but why do you need this `updatedReads` instead of just using the evaluated `samples` from line 76?